agent/exec/container/adapter.go

```diff
@@ -61,15 +62,28 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := rate.NewLimiter(rate.Every(1000*time.Millisecond), 1)
+
 	for {
 		if err := dec.Decode(&m); err != nil {
 			if err == io.EOF {
 				break
 			}
 			return err
 		}
-		// TODO(stevvooe): Report this status somewhere.
-		logrus.Debugln("pull progress", m)
+		if spamLimiter.Allow() {
+			if progress, ok := m["progressDetail"].(map[string]interface{}); ok {
+				// only post pull progress to log every couple of seconds
+				log.G(ctx).WithFields(logrus.Fields{
+					"id":      m["id"],
```

Comment:
No, if the field doesn't exist, you'll get a zero value, so we should be good. In a previous revision, I had nil fields in the `Fields` map and it worked alright, printed out `nil`. The only thing we need to check for is the presence of `m["progressDetail"]` because we have to cast it to a map.